src/builder/QuarkBuilder.sol

```diff
@@ -261,28 +250,38 @@ contract QuarkBuilder {
     }
 
     // Assert that each chain has sufficient funds to cover the max cost for that chain.
-    // NOTE: This check assumes we will not be bridging payment tokens for the user.
+    // NOTE: This check strategy is:
+    // 1. Accrue available fund on each chain (chain balance - max cost)
+    // 2. If total available fund >= transferIntent then pass, otherwise MaxCostTooHigh()
+    // 3. Make sure the transferIntent chain always has enough fund to cover max cost
```

Comment:
I'm confused about what this change is for. It looks like you are no longer checking there is enough payment tokens to cover max payment costs on *all chains*, which seems wrong. Now, it only checks the target chain.

Reply:
Existing logics blindly add transfer.Intent on all the chains for payment need. 
So when user transfer actions that requires bridging, it will duplicated the amount required to transfer, and ended with `maxCostTooHigh()`

When updating the logics, I realized we should just check the chain that takes actions, as checking on all chains on maxCost might fail some actions that not necessary will fail. 

i.e. 
maxCost = $0.1 for Base
Mainnet = $3
Base = $0.05
And user wants to do transfer.  This check will fail when it checks user Base portfolio as $0.05 < $0.1

Reply:
Synced offline, has adjusted wit the new assertPaymentAffordable logics.